test-utils/src/commonMain/kotlin/ai/koog/test/utils/InMemoryFS.kt

```diff
@@ -0,0 +1,62 @@
+package ai.koog.test.utils
+
+import ai.koog.rag.base.files.FileMetadata
+import ai.koog.rag.base.files.FileSystemProvider
+import kotlinx.io.Sink
+import kotlinx.io.Source
+
+/**
+ * Minimal in-memory filesystem for tests. Implements ReadWrite<String> and stores text as ByteArray.
+ * Shared across integration tests to avoid duplication.
+ */
+public class InMemoryFS : FileSystemProvider.ReadWrite<String> {
+    private val files = mutableMapOf<String, ByteArray>()
+    private val directories = mutableSetOf<String>()
+
+    override fun fromAbsolutePathString(path: String): String = path
+    override fun toAbsolutePathString(path: String): String = path
+    override fun joinPath(base: String, vararg parts: String): String = (sequenceOf(base) + parts.asSequence()).joinToString("/")
+    override fun name(path: String): String = path.substringAfterLast('/')
+    override fun parent(path: String): String? = path.substringBeforeLast('/', missingDelimiterValue = "").ifBlank { null }
+    override fun extension(path: String): String = name(path).substringAfterLast('.', "")
+
+    override suspend fun exists(path: String): Boolean = files.containsKey(path) || directories.contains(path)
+    override suspend fun metadata(path: String): FileMetadata? = when {
+        files.containsKey(path) -> FileMetadata(FileMetadata.FileType.File, hidden = false)
+        directories.contains(path) -> FileMetadata(FileMetadata.FileType.Directory, hidden = false)
+        else -> null
+    }
+    override suspend fun size(path: String): Long = files[path]?.size?.toLong() ?: 0L
+
+    override suspend fun readBytes(path: String): ByteArray = files[path] ?: throw IllegalStateException("No such file: $path")
```

Comment:
while it might be not so important in tests, but it should be IllegalArgumentException, see the docs